desc/compute/_qs.py

```diff
@@ -2,131 +2,296 @@
+    dim=1,
+    params=[],
+    transforms={"w": [[0, 0, 1]]},
+    profiles=[],
+    function_of="rtz",
+    data=["w_mn", "G", "I", "iota", "lambda_z"],
+)
+def _nu_z(params, transforms, profiles, data, **kwargs):
+    GI = data["G"] + data["iota"] * data["I"]
+    w_z = transforms["w"].transform(data["w_mn"], dr=0, dt=0, dz=1)
+    data["nu_z"] = (w_z - data["I"] * data["lambda_z"]) / GI
+    return data
+
+
+@register_compute_fun(
+    name="theta_B",
+    label="\\theta_{B}",
+    units="rad",
+    units_long="radians",
+    description="Boozer poloidal angular coordinate",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    function_of="rtz",
+    data=["theta", "lambda", "iota", "nu"],
+)
+def _theta_B(params, transforms, profiles, data, **kwargs):
+    data["theta_B"] = data["theta"] + data["lambda"] + data["iota"] * data["nu"]
+    return data
+
+
+@register_compute_fun(
+    name="zeta_B",
+    label="\\zeta_{B}",
+    units="rad",
+    units_long="radians",
+    description="Boozer toroidal angular coordinate",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    function_of="rtz",
+    data=["zeta", "nu"],
+)
+def _zeta_B(params, transforms, profiles, data, **kwargs):
+    data["zeta_B"] = data["zeta"] + data["nu"]
+    return data
+
+
+@register_compute_fun(
+    name="sqrt(g)_B",
+    label="\\sqrt{g}_{B}",
+    units="~",
+    units_long="None",
+    description="Jacobian determinant of Boozer coordinates",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    function_of="rtz",
+    data=["lambda_t", "lambda_z", "nu_t", "nu_z", "iota"],
+)
+def _sqrtg_B(params, transforms, profiles, data, **kwargs):
+    data["sqrt(g)_B"] = (1 + data["lambda_t"]) * (1 + data["nu_z"]) + (
+        data["iota"] - data["lambda_z"]
+    ) * data["nu_t"]
+    return data
+
+
+@register_compute_fun(
+    name="|B|_mn",
+    label="B_{mn}^{Boozer}",
+    units="T",
+    units_long="Tesla",
+    description="Boozer harmonics of magnetic field",
+    dim=1,
+    params=[],
+    transforms={"B": [[0, 0, 0]]},
+    profiles=[],
+    function_of="rtz",
+    data=["sqrt(g)_B", "|B|", "rho", "theta_B", "zeta_B"],
+)
+def _B_mn(params, transforms, profiles, data, **kwargs):
+    nodes = jnp.array([data["rho"], data["theta_B"], data["zeta_B"]]).T
+    norm = 2 ** (3 - jnp.sum((transforms["B"].basis.modes == 0), axis=1))
+    data["|B|_mn"] = (
+        norm  # 1 if m=n=0, 2 if m=0 or n=0, 4 if m!=0 and n!=0
+        * jnp.matmul(
+            transforms["B"].basis.evaluate(nodes).T, data["sqrt(g)_B"] * data["|B|"]
+        )
+        / transforms["B"].grid.num_nodes
     )
+    return data
+
+
+@register_compute_fun(
+    name="B modes",
+    label="Boozer modes",
+    units="~",
+    units_long="None",
+    description="Boozer harmonics",
+    dim=1,
+    params=[],
+    transforms={"B": [[0, 0, 0]]},
+    profiles=[],
+    function_of="rtz",
+    data=["sqrt(g)_B", "|B|", "rho", "theta_B", "zeta_B"],
```

Comment:
Does it actually need this data? I don't think it should have any data dependencies.